spec/system/course_progress_badge_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'Course Progress Badge', type: :system do
+  let!(:path) { create(:path, default_path: true) }
+  let!(:course) { create(:course, path: path) }
+  let!(:section) { create(:section, course: course) }
+  let!(:first_lesson) { create(:lesson, section: section) }
+  let!(:second_lesson) { create(:lesson, section: section) }
+  let!(:user) { create(:user) }
+
+  before do
+    sign_in(user)
+  end
+
+  context 'when course has not been started' do
+    it 'does not show progress percentage' do
+      visit path_course_path(path, course)
+
+      within find(:test_id, 'default-badge') do
+        expect(page).not_to have_content('0%')
+        expect(page).not_to have_content('COMPLETE')
+      end
+    end
+  end
+
+  context 'when course has some progress' do
+    it 'shows percentage of completion' do
+      visit path_course_lesson_path(path, course, first_lesson)
```

Comment:
Great idea testing this by clicking on the complete button on the course page instead of the dashboard page 👏 it tests the dynamic behaviour of the badge getting updated on the fly.

Reply:
You think too highly of me... This is actually clicking on the complete button on the lesson page, not the course page. It does make the most sense to use the course page, but I'll have to refactoring the naming of the buttons to be dynamic with the lesson name, because as it stands right now all the buttons are named 'lesson_complete_btn'.

Reply:
ahh I just seen course in the route helper and instantly assumed it was the course page 😆 

 The lesson page is perfectly fine though, it has the exact same behaviour with the complete button and the badge updating.